src/main/webapp/resources/js/pages/search/SearchLayout.tsx

```diff
@@ -0,0 +1,312 @@
+};
+
+export type SearchProject = SearchItem & {
+  samples: SearchSample[];
+};
+
+export type SearchSample = SearchItem & {
+  projects: SearchProject[];
+};
+
+export type SampleTableType = Pick<
+  Sample,
+  "name" | "organism" | "projects" | "createdDate" | "modifiedDate"
+> & {
+  key: "string";
+};
+
+export interface TableParams {
+  pagination?: TablePaginationConfig;
+  sortField?: string;
+  sortOrder?: string;
+  filters?: Record<string, FilterValue>;
+  columnKey?: string;
+  order: "ascend" | "descend";
+}
+
+const initial_table_params = JSON.stringify({
+  pagination: {
+    current: 1,
+    pageSize: 10,
+  },
+});
+
+/**
+ * React component to layout and handle events for the global search
+ * @constructor
+ */
+export default function SearchLayout() {
+  const user = useLoaderData() as CurrentUser;
+  const [searchParams, setSearchParams] = useSearchParams();
+  const debouncedSetSearchParams = debounce(async (value) => {
+    setSearchParams(value);
+  }, 500);
+  const [type, setType] = useState<SearchType>("projects");
+  const [global, setGlobal] = useState<boolean>(user.admin);
+
+  const [projects, setProjects] = useState<{
+    content: SearchProject[];
+    total: number;
+  }>();
+  const [projectsTableParams, setProjectsTableParams] = useState<TableParams>(
+    JSON.parse(initial_table_params)
+  );
+  const debouncedSetProjectTableParams = debounce(async (params) => {
+    setProjectsTableParams(params);
+  }, 300);
+
+  const [samples, setSamples] = useState<{
+    content: SearchSample[];
+    total: number;
+  }>();
+  const [samplesTableParams, setSamplesTableParams] = useState<TableParams>(
+    JSON.parse(initial_table_params)
+  );
+  const debouncedSetSamplesTableParams = debounce(async (params) => {
+    setSamplesTableParams(params);
+  }, 300);
+
+  const handleProjectsTableChange: TableProps<SearchProject>["onChange"] = (
+    pagination,
+    filters,
+    sorter
+  ): void => {
+    debouncedSetProjectTableParams({
+      pagination,
+      filters,
+      ...sorter,
+    });
+  };
+
+  const handleSamplesTableChange: TableProps<SearchSample>["onChange"] = (
+    pagination,
+    filters,
+    sorter
+  ): void => {
+    debouncedSetSamplesTableParams({
+      pagination,
+      filters,
+      ...sorter,
+    });
+  };
+
+  const fetchData = useCallback(() => {
+    const fetchSamples = async () =>
+      axios.post(setBaseUrl(`/ajax/search/samples`), {
+        global,
+        pagination: samplesTableParams.pagination,
+        order: [
+          {
+            property: samplesTableParams.columnKey || `sampleName`,
+            direction: samplesTableParams.order === "ascend" ? `asc` : `desc`,
+          },
+        ],
+        search: [
+          {
+            property: `name`,
+            value: searchParams.get("query"),
+            operation: "MATCH_IN",
+          },
+        ],
+      });
+
+    const fetchProjects = async () =>
+      axios.post(setBaseUrl(`/ajax/search/projects`), {
```

Comment:
Can we keep api calls in the file for it in the api directory instead?

Reply:
Updated in 89edf7e